### PR TITLE
Add API docs, unit-tests and usage examples

### PR DESCRIPTION
**Pull request by dwillmer**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -48,6 +51,31 @@ function overrideCursor(cursor: string): IDisposable {
 
 /**
  * Test whether a client position lies within a node.
+ *
+ * @param node - The HTMLElement on which to test against.
+ *
+ * @param x - The x co-ordinate of the position to hit test.
```
Comment (blink1073) [inline src/index.ts]: *coordinate

Comment (dwillmer) [inline src/index.ts]: dictionary.reference.com/browse/coordinate

File: src/index.ts [hunk 2]
```diff
@@ -48,6 +51,31 @@ function overrideCursor(cursor: string): IDisposable {
...
+ *
+ * #### Example
+ * ```typescript
+ * var obj = document.createElement('img');
+ * obj.style.position = 'absolute';
+ * obj.style.left = 0;
+ * obj.style.top = 0;
+ * obj.width = 100;
+ * obj.height = 100;
+ * document.body.appendChild( obj );
```
Comment (blink1073) [inline src/index.ts]: What's with all the ` `?

Comment (dwillmer) [inline src/index.ts]: Fixed, sorry!

File: src/index.ts [hunk 3]
```diff
@@ -48,6 +51,31 @@ function overrideCursor(cursor: string): IDisposable {
...
+ * ```typescript
+ * var obj = document.createElement('img');
+ * obj.style.position = 'absolute';
+ * obj.style.left = 0;
+ * obj.style.top = 0;
+ * obj.width = 100;
+ * obj.height = 100;
+ * document.body.appendChild(obj);
+ *
+ * hitTest( obj, 50, 50 ); // true
```
Comment (blink1073) [inline src/index.ts]: More extraneous spaces.

Comment (dwillmer) [inline src/index.ts]: Fixed.

**Comment by blink1073** [comment]
Nice:

```
SUMMARY:
✔ 6 tests completed
-----------|----------|----------|----------|----------|----------------|
File       |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
-----------|----------|----------|----------|----------|----------------|
 lib/      |      100 |       70 |      100 |      100 |                |
  index.js |      100 |       70 |      100 |      100 |                |
-----------|----------|----------|----------|----------|----------------|
All files  |      100 |       70 |      100 |      100 |                |
-----------|----------|----------|----------|----------|----------------|
```

File: src/index.ts (hunk 1 above)
Comment (sccolbert) [inline src/index.ts]: *burn*

**Pull request merged**